### ISSUE TITLE
[a11y]: <fieldset> elements must have a single, non-empty <legend> as a label

### ISSUE THREAD

**Issue by preranagiri** [carbon-design-system/carbon#12886]
### Package

carbon-components-react

### Browser

Firefox

### Operating System

MacOS

### Package version

7.57.4

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

We use RadioButton in our application and on running the IBM Equal Access Accessibility Checker we can see the following accessibility issue

- <fieldset> elements must have a single, non-empty <legend> as a label. There is AVT1 issue raised against this.

element location:

```
<fieldset
    class="bx--radio-button-group bx--radio-button-group--label-right">
```

<img width="1519" alt="image" src="https://user-images.githubusercontent.com/13781018/208675102-fae5f263-2cec-4fbb-a30d-9d2c3f339a0f.png">

### Reproduction/example

N/A

### Steps to reproduce

N/A

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Hey there @preranagiri , do you have a link to a CodeSandbox so we can take a look at a simplified reproduction? I am unable to reproduce the violation in our [storybook](https://react.carbondesignsystem.com/iframe.html?args=&id=components-radiobutton--default&viewMode=story), so you could be missing a prop.

Here is the storybook code as an example:

```jsx
<RadioButtonGroup
      legendText="Group label"
      name="radio-button-group"
      defaultSelected="radio-1">
      <RadioButton
        labelText="Radio button label"
        value="radio-1"
        id="radio-1"
      />
      <RadioButton
        labelText="Radio button label"
        value="radio-2"
        id="radio-2"
      />
      <RadioButton
        labelText="Radio button label"
        value="radio-3"
        id="radio-3"
        disabled
      />
 </RadioButtonGroup>
```
